genie-core/src/main/java/com/netflix/genie/core/jpa/services/JpaJobSearchServiceImpl.java

```diff
@@ -270,14 +271,14 @@ public Job getJob(
     @Override
     public JobStatus getJobStatus(@NotBlank final String id) throws GenieException {
         log.debug("Called with id {}", id);
-        if (this.jobRepository.exists(id)) {
-            final CriteriaBuilder cb = this.entityManager.getCriteriaBuilder();
-            final CriteriaQuery<JobStatus> query = cb.createQuery(JobStatus.class);
-            final Root<JobEntity> root = query.from(JobEntity.class);
-            query.select(root.get(JobEntity_.status));
-            query.where(cb.equal(root.get(JobEntity_.id), id));
-            return this.entityManager.createQuery(query).getSingleResult();
-        } else {
+        final CriteriaBuilder cb = entityManager.getCriteriaBuilder();
```

Comment:
Since this is much less dynamic than some of our other search queries did you see if theres a way to use a repository or named query which is more static for pulling back the same info rather than building it up every time? Can we make this static somehow i guess is the core question